src/main/ipv-stub/ipv-authorize.ts

```diff
@@ -15,38 +17,98 @@ export const handler: Handler = async (
     case "POST":
       return post(event);
     default:
-      logger.error("Method not allowed");
-      return {
-        statusCode: 405,
-        body: JSON.stringify({
-          message: "Method not allowed",
-        }),
-      };
+      return errorResponse(405, "Method not allowed");
   }
 };
 
-function get(_event: APIGatewayProxyEvent): APIGatewayProxyResult {
+async function get(
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> {
   logger.info("Reached the GET endpoint");
+
+  const headers: APIGatewayProxyEventHeaders = event.headers;
+  const bearerToken = headers["Authorization"];
+  if (!bearerToken) {
+    return errorResponse(400, "Bearer token not found");
+  }
+  const encryptedJwt = bearerToken.split(" ")[1];
+
+  const ipvPrivateKeyPem = process.env.IPV_PRIVATE_KEY;
+  if (!ipvPrivateKeyPem) {
+    return errorResponse(500, "Private key not found");
+  }
+  const privateKey = await importPKCS8(ipvPrivateKeyPem, "RSA-OAEP-256");
+
+  let encodedJwt = "";
+  try {
+    const { plaintext } = await compactDecrypt(encryptedJwt, privateKey);
+    encodedJwt = plaintext.toString();
+  } catch (error) {
+    return errorResponse(500, "Decryption failed");
+  }
+
+  const parts = encodedJwt.split(".");
+  if (parts.length !== 3) {
+    return errorResponse(400, "Decrypted JWT is in invalid format");
+  }
+
+  let decodedHeader, decodedPayload, decodedSignature;
+  try {
+    [decodedHeader, decodedPayload, decodedSignature] = parts.map((part) =>
+      Buffer.from(part, "base64url").toString("utf8")
+    );
+  } catch (error) {
+    return errorResponse(500, "Decoding failed");
+  }
```

Comment:
Are we planning to validate the signature here?

Reply:
Have looked at this but couldn't get it working immediately - don't want to spend too much time as it's not essential. Will create a ticket and address later on if we decide to